index.js

```diff
@@ -0,0 +1,45 @@
+import { preprocess } from './lib/preprocess.js'
+import { evaluate } from './lib/evaluate.js'
+
+export const startEvaluate = ({
+  ieContract,
+  ieContractWithSigner,
+  web3Storage,
+  logger
+}) => {
+  const rounds = {}
+  const cidsSeen = []
+  const roundsSeen = []
+
+  // Listen for events
+  ieContract.on('MeasurementsAdded', (cid, _roundIndex) => {
+    const roundIndex = Number(_roundIndex)
+    if (cidsSeen.includes(cid)) return
+    cidsSeen.push(cid)
+
+    console.log('Event: MeasurementsAdded', { roundIndex })
+    // Preprocess
+    preprocess({
+      rounds,
+      cid,
+      roundIndex,
+      web3Storage,
+      logger
+    }).catch(console.error)
+  })
+
+  ieContract.on('RoundStart', _roundIndex => {
+    const roundIndex = Number(_roundIndex)
+    if (roundsSeen.includes(roundIndex)) return
+    roundsSeen.push(roundIndex)
```

Comment:
Ditto. 

Can we limit the size of this array? Is it perhaps enough to remember the last round seen?

What happens after the service restarts?

Reply:
See above